### PR TITLE
[FLINK-23241][docs-zh] Translate the page of "Working with State " in…

### PR DESCRIPTION
**Pull request by zuoniduimian**
What is the purpose of the change
Translate the page of "Working with State " into Chinese

The page url of "Working with State" is https://ci.apache.org/projects/flink/flink-docs-release-1.13/zh/docs/dev/datastream/fault-tolerance/state/.

The markdown file can be found in flink/docs/content.zh/docs/dev/datastream/fault-tolerance/state.md in English.

Brief change log
Translate 'flink/docs/content.zh/docs/dev/datastream/fault-tolerance/state.md'.
Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

Does this pull request potentially affect one of the following parts:
Dependencies (does it add or upgrade a dependency): no
The public API, i.e., is any changed class annotated with @Public(Evolving): no
The serializers: no
The runtime per-record code paths (performance sensitive): no
Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
The S3 file system connector: no
Documentation
Does this pull request introduce a new feature? no
If yes, how is the feature documented? not documented

### CODE REVIEW

**Comment by zuoniduimian** [comment]
@edmondsky @95chenjz  hello,guys,please help me review my translated page,thanks for your efforts!

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 1]
```diff
@@ -25,32 +25,20 @@ specific language governing permissions and limitations
...
 -->
 
-# Working with State
+# 带有状态的处理
 
-In this section you will learn about the APIs that Flink provides for writing
-stateful programs. Please take a look at [Stateful Stream
-Processing]({{< ref "docs/concepts/stateful-stream-processing" >}})
-to learn about the concepts behind stateful stream processing.
+在本节中，你可以了解Flink提供的为有状态编程开发的的API。 请查看 [Stateful Stream
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:30]: ”Flink“和 ”API“前面没加空格

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 2]
```diff
@@ -25,32 +25,20 @@ specific language governing permissions and limitations
...
+Processing]({{< ref "docs/concepts/stateful-stream-processing" >}})，以便了解有状态流处理的概念。
 
 ## Keyed DataStream
 
-If you want to use keyed state, you first need to specify a key on a
-`DataStream` that should be used to partition the state (and also the records
-in the stream themselves). You can specify a key using `keyBy(KeySelector)`
-in Java/Scala API or `key_by(KeySelector)` in Python API on a `DataStream`.
-This will yield a `KeyedStream`, which then allows operations that use keyed state.
+如果要使用 keyed state,，首先需要在 `DataStream` 中指定用于为状态（以及流本身中的记录）分区的 key。可以在 `DataStream` 上使用 Java/Scala API 中的 `keyby（keyselector）` 或 Python API 中的 `Key_by（keyselector）` 指定 key。这样会产生一个 `keyedStream` ，在这个数据流上支持使用 keyed state 的操作。
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:35]: 这里表示 API 括号你好像都写成中文的了，比如 `keyby（keyselector）` 和 `Key_by（keyselector）`，既然是 API，应该用英文括号

Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:35]: 这里表示 API 括号你好像都写成中文的了，比如 `keyby（keyselector）` 和 `Key_by（keyselector）`，既然是 API，应该用英文括号

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 3]
```diff
@@ -85,15 +73,9 @@ keyed = words.key_by(lambda row: row[0])
 {{< /tab >}}
 {{< /tabs >}}
 
-#### Tuple Keys and Expression Keys
+#### 元祖键和表达式键
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:76]: 这里在括号里指明英文单词是不是更好一点，如：“元祖（Tuple）键和表达式键”

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md (hunk 1 above)
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:30]: ”Flink“和 ”API“前面没加空格

**Review by RollsBean — COMMENTED** [review]
我也是刚开始为社区翻译文档，所以只是提了一些格式上的问题，其他问题可能还是需要别人再 review 一下了

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 4]
```diff
@@ -24,28 +24,23 @@ specific language governing permissions and limitations
...
+应用程序参数处理
 -------------------------------
-Almost all Flink applications, both batch and streaming, rely on external configuration parameters.
-They are used to specify input and output sources (like paths or addresses), system parameters (parallelism, runtime configuration), and application specific parameters (typically used within user functions).
+几乎所有的Flink应用程序，也就是批和流程序，都依赖于外部配置参数。这些配置参数指定输入和输出源（如路径或地址），系统参数（并行度，运行时配置）和应用程序特定参数（通常在用户函数中使用）。
 
-Flink provides a simple utility called `ParameterTool` to provide some basic tooling for solving these problems.
-Please note that you don't have to use the `ParameterTool` described here. Other frameworks such as [Commons CLI](https://commons.apache.org/proper/commons-cli/) and
-[argparse4j](http://argparse4j.sourceforge.net/) also work well with Flink.
+Flink提供一个名为 `Parametertool` 的简单实用类，为解决以上问题提供了基本的工具。 这里请注意，此处描述的` parametertool` 并不是必须的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/)等其他框架也与Flink兼容非常好。
```
[the remark this answers is not in the document]
Comment (zuoniduimian) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:33]: thanks for your comment.I will modify it

**Comment by zuoniduimian** [comment]
@edmondsky @95chenjz @RollsBean hello,guys,I have modified the translated page in question.If you are spare ,please help me review my translated page,thanks for your efforts!

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 5]
```diff
@@ -117,7 +116,7 @@ ParameterTool parameters = ParameterTool.fromArgs(args);
 
 // set up the execution environment
 final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
-env.getConfig().setGlobalJobParameters(parameters);
+        env.getConfig().setGlobalJobParameters(parameters);
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:119]: 这个地方多改了吧， env.getConfig() 前面多了 tab

**Review by RollsBean — COMMENTED** [review]
我看你提交了两个文档的翻译，一般一个 PR 只提交一个 JIRA 的修改，这个你可以拆成两个 PR 来提交，而且一般都是从你的开发分支合并，我看你直接从 master 提的 PR

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 6]
```diff
@@ -28,18 +28,18 @@ under the License.
 
 应用程序参数处理
 -------------------------------
-几乎所有的Flink应用程序，也就是批和流程序，都依赖于外部配置参数。这些配置参数指定输入和输出源（如路径或地址），系统参数（并行度，运行时配置）和应用程序特定参数（通常在用户函数中使用）。
+几乎所有的 Flink 应用程序，也就是批和流程序，都依赖于外部配置参数。这些配置参数指定输入和输出源（如路径或地址），系统参数（并行度，运行时配置）和应用程序特定参数（通常在用户函数中使用）。
 
-Flink提供一个名为 `Parametertool` 的简单实用类，为解决以上问题提供了基本的工具。 这里请注意，此处描述的` parametertool` 并不是必须的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/)等其他框架也与Flink兼容非常好。
+Flink 提供一个名为 `Parametertool` 的简单实用类，为解决以上问题提供了基本的工具。请注意，此处描述的 `Parametertool` 并不是必须的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/) 等其他框架也可以非常好地兼容 Flink。
```
Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:33]:
```suggestion
为解决以上问题Flink 提供一个名为 `Parametertool` 的简单公共类，其中包含了一些基本的工具。请注意，这里说的 `Parametertool` 并不是必须使用的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/) 等其他框架也可以非常好地兼容 Flink。
```

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 7]
```diff
@@ -28,18 +28,18 @@ under the License.
...
 
 ### `ParameterTool` 读取配置值
 
-`ParameterTool` 提供一组预定义的静态方法，用于读取配置信息。 该工具类内部实现倾向于使用` Map<string，string>`，因此很容易与我们自己的配置集成。
+`ParameterTool` 提供一组预定义的静态方法，用于读取配置信息。该工具类内部实现倾向于使用 `Map<string，string>`，因此很容易与我们自己的配置集成。
```
Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:37]:
```suggestion
`ParameterTool` 已经定义好了一组静态方法，用于读取配置信息。该工具类内部使用了 `Map<string，string>`类型，这样使得它可能很容易的与你的配置集成在一起。
```

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 8]
```diff
@@ -52,7 +52,6 @@ InputStream propertiesFileInputStream = new FileInputStream(file);
 ParameterTool parameter = ParameterTool.fromPropertiesFile(propertiesFileInputStream);
 ```
 
-
 #### 配置值来自命令行
 
 该操作从命令行获取像 `--input hdfs:///mydata --elements 42` 的参数。
```
Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:57]:
```suggestion
以下方法可以从命令行中获取参数，如 `--input hdfs:///mydata --elements 42` 。
```

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 9]
```diff
@@ -66,19 +65,19 @@ public static void main(String[] args) {
...
 
-启动JVM时，可以将系统属性传递给JVM：`-Dinput=hdfs:///mydata`。还可以从这些系统属性初始化 `ParameterTool`：
+启动 JVM 时，可以将系统属性传递给 JVM ：`-Dinput=hdfs:///mydata` 。你也可以从这些系统属性初始化 `ParameterTool`：
 
 ```java
 ParameterTool parameter = ParameterTool.fromSystemProperties();
 ```
 
-### Flink程序中使用参数
+### Flink 程序中使用参数
```
Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:74]:
```suggestion
### 在Flink程序中使用参数
```

**Comment by zuoniduimian** [comment]
@edmondsky @95chenjz  @RollsBean   I have made some changes according to your suggestions，please help me review these changes.Thanks!

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 10]
```diff
@@ -27,18 +27,18 @@ under the License.
...
 ## Keyed DataStream
 
-如果要使用 keyed state ，首先需要在 `DataStream` 中指定用于为状态（以及流本身中的记录）分区的 key 。你可以在 `DataStream` 上使用 Java/Scala API 中的 `keyby(Keyselector)` 或 Python API 中的 `key_by(Keyselector)` 指定 key 。然后这样会产生一个 `KeyedStream` ，支持使用 keyed state 的操作。
+如果你想使用 keyed state ，首先需要在 `DataStream` 中指定用于为状态（以及流本身中的记录）分区的 key 。你可以在 `DataStream` 上使用 Java/Scala API 中的 `keyby(Keyselector)` 或 Python API 中的 `key_by(Keyselector)` 指定 key 。然后这样会产生一个支持使用 keyed state 操作的 `KeyedStream`。
 
-一个 key 选择器将单个记录作为输入并返回记录的 key 。这个 key 可以是任意类型的，并且**必须**取自确定计算结果。
+key 选择器将单个记录作为输入并返回记录的 key。这个 key 可以是任意类型的，并且**必须**取自确定计算结果。
 
-Flink 的数据模型并不是基于键值对的。因此，不需要将数据集类型硬性地包装为键值对。这里的 keys 是“虚拟”的：keys 可以理解为真实数据上的函数，用于分组算子计算。
+Flink 的数据模型并不是基于键值对的。因此，不需要将数据集类型硬性地包装为键值对。这里的 keys 是"虚拟"的：keys 可以理解为真实数据上的函数，用于分组算子计算。
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:39]: “虚拟”，中文里面使用中文标点符号

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 11]
```diff
@@ -28,13 +28,13 @@ under the License.
...
-几乎所有的 Flink 应用程序，也就是批和流程序，都依赖于外部配置参数。这些配置参数指定输入和输出源（如路径或地址），系统参数（并行度，运行时配置）和应用程序特定参数（通常在用户函数中使用）。
+几乎所有的批和流的 Flink 应用程序，都依赖于外部配置参数。这些配置参数可以用于指定输入和输出源（如路径或地址）、系统参数（并行度，运行时配置）和特定的应用程序参数（通常使用在用户自定义函数）。
 
-Flink 提供一个名为 `Parametertool` 的简单实用类，为解决以上问题提供了基本的工具。请注意，此处描述的 `Parametertool` 并不是必须的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/) 等其他框架也可以非常好地兼容 Flink。
+为解决以上问题，Flink 提供一个名为 `Parametertool` 的简单公共类，其中包含了一些基本的工具。请注意，这里说的 `Parametertool` 并不是必须使用的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/) 等其他框架也可以非常好地兼容 Flink。
 
 ### `ParameterTool` 读取配置值
 
-`ParameterTool` 提供一组预定义的静态方法，用于读取配置信息。该工具类内部实现倾向于使用 `Map<string，string>`，因此很容易与我们自己的配置集成。
+`ParameterTool` 已经定义好了一组静态方法，用于读取配置信息。该工具类内部使用了 `Map<string，string>` 类型，这样使得它可能很容易的与你的配置集成在一起。
```
Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:37]: sugg

Comment (edmondsky) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:37]:
```suggestion
`ParameterTool` 已经定义好了一组静态方法，用于读取配置信息。该工具类内部使用了 `Map<string，string>` 类型，这样使得它可以很容易的与你的配置集成在一起。
```

**Comment by zuoniduimian** [comment]
@wuchong  @xccui @klion26  @PatrickRen @becketqin   hello,guys,Could you review my PR in your spare time? looking for your feedback.Thanks for your efforts!

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 12]
```diff
@@ -24,28 +24,23 @@ specific language governing permissions and limitations
...
 
-Flink provides a simple utility called `ParameterTool` to provide some basic tooling for solving these problems.
-Please note that you don't have to use the `ParameterTool` described here. Other frameworks such as [Commons CLI](https://commons.apache.org/proper/commons-cli/) and
-[argparse4j](http://argparse4j.sourceforge.net/) also work well with Flink.
+为解决以上问题，Flink 提供一个名为 `Parametertool` 的简单公共类，其中包含了一些基本的工具。请注意，这里说的 `Parametertool` 并不是必须使用的。[Commons CLI](https://commons.apache.org/proper/commons-cli/) 和 [argparse4j](http://argparse4j.sourceforge.net/) 等其他框架也可以非常好地兼容 Flink。
 
+### `ParameterTool` 读取配置值
 
-### Getting your configuration values into the `ParameterTool`
+`ParameterTool` 已经定义好了一组静态方法，用于读取配置信息。该工具类内部使用了 `Map<string，string>` 类型，这样使得它可以很容易的与你的配置集成在一起。
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:37]:
```suggestion
`ParameterTool` 已经定义好了一组静态方法，用于读取配置信息。该工具类内部使用了 `Map<string，string>` 类型，这样使得它可以很容易地与你的配置集成在一起。
```

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 13]
```diff
@@ -93,29 +88,28 @@ parameter.getNumberOfParameters()
...
 ```
 
-and then use it inside the function for getting values from the command line.
+然后在函数内使用它以获取命令行的传递的参数。
 
-#### Register the parameters globally
+#### 全局注册参数
 
-Parameters registered as global job parameters in the `ExecutionConfig` can be accessed as configuration values from the JobManager web interface and in all functions defined by the user.
+从 JobManager web 界面和用户定义的所有函数中可以以配置值的方式访问在 `ExecutionConfig` 中注册为全局作业参数。
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:110]:
```suggestion
从 JobManager web 界面和所有用户定义的函数中可以以配置值的方式访问在 `ExecutionConfig` 中注册的全局作业参数。
```

File: docs/content.zh/docs/dev/datastream/application_parameters.md [hunk 14]
```diff
@@ -124,18 +118,17 @@ ParameterTool parameters = ParameterTool.fromArgs(args);
...
 public static final class Tokenizer extends RichFlatMapFunction<String, Tuple2<String, Integer>> {
 
     @Override
     public void flatMap(String value, Collector<Tuple2<String, Integer>> out) {
-	ParameterTool parameters = (ParameterTool)
-	    getRuntimeContext().getExecutionConfig().getGlobalJobParameters();
-	parameters.getRequired("input");
-	// .. do more ..
+        ParameterTool parameters = (ParameterTool)
+                getRuntimeContext().getExecutionConfig().getGlobalJobParameters();
```
Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:129]: 这里多了一行，你用 md 编辑器看一下渲染后的效果

Comment (RollsBean) [inline docs/content.zh/docs/dev/datastream/application_parameters.md:129]: 这里多了一行，你用 md 编辑器查看一下

**Review by RollsBean — CHANGES_REQUESTED** [review]
Left some suggestions

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 13 above)
Comment (zuoniduimian) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:110]: Hello, I think there is no problem with the translation of this attribute.

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 14 above)
Comment (zuoniduimian) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:129]: Hello, I'm consistent with the original text here

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 13 above)
Comment (RollsBean) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:110]: ...可以/   以...的方式   /   访问（动词）  / 在  `ExecutionConfig` 中注册为全局作业参数（定语） / 

你看下上面句子结构，完全不通顺呀，访问后面都没有名词（宾语）

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 14 above)
Comment (RollsBean) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:129]: image
这里明明格式有变动

Comment (RollsBean) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:129]: 这个不知道为啥显示有空格，你本地跑下，格式没错就行了

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 13 above)
Comment (zuoniduimian) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:110]: ok.

File: docs/content.zh/docs/dev/datastream/application_parameters.md (hunk 14 above)
Comment (zuoniduimian) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:129]: There's really no problem here

Comment (RollsBean) [reply docs/content.zh/docs/dev/datastream/application_parameters.md:129]: ok

**Review by RollsBean — APPROVED** [review]

**Comment by zuoniduimian** [comment]
@wuchong  @xccui  @klion26  @PatrickRen  @becketqin  hello,guys,Could you review my PR in your spare time? looking for your feedback.Thanks for your efforts!

**Comment by zuoniduimian** [comment]
@PatrickRen  hello,Could you help me  review my PR in your spare time? thank you !

**Pull request closed without merging**